scripts/verify-handoff.sh

```
@@ -0,0 +1,105 @@
+[ "$ELAPSED" -lt "$TIMEOUT" ] || fail "successor daemon did not come up within ${TIMEOUT}s"
+
+log "step 4/5: capturing post-restart state"
+AFTER="$(pids_snapshot)"
+AFTER_ATTEMPTED="$(handoff_counter attempted)"
+AFTER_TRANSFERRED="$(handoff_counter transferred)"
+AFTER_FALLBACK="$(handoff_counter fallback)"
+AFTER_COUNT="$(printf '%s\n' "$AFTER" | wc -l | tr -d ' ')"
+log "  observed $AFTER_COUNT upstream(s); post counters: attempted=$AFTER_ATTEMPTED transferred=$AFTER_TRANSFERRED fallback=$AFTER_FALLBACK"
+
+log "step 5/5: comparing PID sets"
+DIFF="$(diff -u <(printf '%s\n' "$BEFORE") <(printf '%s\n' "$AFTER") || true)"
```

Shellcheck (0.11.0)

[warning] In POSIX sh, process substitution is undefined.

(SC3001)